python_docs_theme/static/copybutton.js

```diff
@@ -1,65 +1,47 @@
-            // span element so that we can control visibility
-            if (el.nodeType === Node.TEXT_NODE && el.textContent.trim()) {
-                const wrapper = document.createElement("span")
-                el.after(wrapper)
-                wrapper.appendChild(el)
-                el = wrapper
-            }
-            yield el
+// Extract copyable text from the code block ignoring the
+// prompts and output.
+function getCopyableText(rootElement) {
+    rootElement = rootElement.cloneNode(true)
+    // tracebacks (.gt) contain bare text elements that
+    // need to be removed
+    const tracebacks = rootElement.querySelectorAll(".gt")
+    for (const el of tracebacks) {
+        while (
+            el.nextSibling &&
+            (el.nextSibling.nodeType !== Node.DOCUMENT_NODE ||
+                !el.nextSibling.matches(".gp, .go"))
+        ) {
+            el.nextSibling.remove()
         }
     }
+    // Remove all elements with the "go" (Generic.Output),
+    // "gp" (Generic.Prompt), or "gt" (Generic.Traceback) CSS class
+    const elements = rootElement.querySelectorAll(".gp, .go, .gt")
+    for (const el of elements) {
+        el.remove()
+    }
+    return rootElement.innerText.trim()
 }
 
-
 const loadCopyButton = () => {
-    /* Add a [>>>] button in the top-right corner of code samples to hide
-     * the >>> and ... prompts and the output and thus make the code
-     * copyable. */
-    const hide_text = _("Hide the prompts and output")
-    const show_text = _("Show the prompts and output")
-
-    const button = document.createElement("span")
+    const button = document.createElement("button")
     button.classList.add("copybutton")
-    button.innerText = ">>>"
-    button.title = hide_text
-    button.dataset.hidden = "false"
-    const buttonClick = event => {
+    button.type = "button"
+    button.innerText = _("Copy")
+    button.title = _("Copy to clipboard")
+
+    const makeOnButtonClick = () => {
+        let timeout = null
         // define the behavior of the button when it's clicked
-        event.preventDefault()
-        const buttonEl = event.currentTarget
-        const codeEl = buttonEl.nextElementSibling
-        if (buttonEl.dataset.hidden === "false") {
-            // hide the code output
-            for (const el of getHideableCopyButtonElements(codeEl)) {
-                el.hidden = true
-            }
-            buttonEl.title = show_text
-            buttonEl.dataset.hidden = "true"
-        } else {
-            // show the code output
-            for (const el of getHideableCopyButtonElements(codeEl)) {
-                el.hidden = false
-            }
-            buttonEl.title = hide_text
-            buttonEl.dataset.hidden = "false"
+        return event => {
+          clearTimeout(timeout)
+          const buttonEl = event.currentTarget
+          const codeEl = buttonEl.nextElementSibling
+          navigator.clipboard.writeText(getCopyableText(codeEl))
```

Comment:
My IDE gives a warning that the returned promise is ignored:

<img width="670" alt="image" src="https://github.com/user-attachments/assets/50e7a186-d820-4b2b-a9b0-bb95759cf2f3" />

Should we have a `.then()` and only do "Copied!" in there?

https://web.dev/patterns/clipboard/copy-text

And maybe also log any error, in case it happens?

Reply:
Thanks for testing it!

> They also use aria-labels for accessibility.

I think that won't be necessary in our case? We're using a regular button with `type="button"` and text content.

> But what happens on a browser without support?

It's gonna raise inside the click handler which shouldn't break anything else but I'll add some proper feature detection.

> Should we have a .then() and only do "Copied!" in there?

Yep, that sounds like a safer option!